### PR TITLE
feat: add a way to abort exports

### PR DESCRIPTION
**Pull request by tiina303**
## Problem

For various reasons export jobs can get stuck in bad states.
We've sometimes used the workaround to delete the plugin config, but that also stops the running update and loses all the history - obviously not great.

## Changes

This allows us to add an abortMessage in Postgres to the export parameters to a running historical export job, which would stop it.

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: plugin-server/src/worker/vm/upgrades/historical-export/export-historical-events-v2.ts [hunk 1]
```diff
@@ -252,6 +253,12 @@ export function addHistoricalEventsExportCapabilityV2(
             return
         }
 
+        if (params.abortMessage) {
+            // For manually triggering the export to abort
+            await stopExport(params, `Export aborted ${params.abortMessage}`, 'fail')
```
Comment (xvello) [inline plugin-server/src/worker/vm/upgrades/historical-export/export-historical-events-v2.ts:258]:
```suggestion
            await stopExport(params, `Export stopped by instance operator: ${params.abortMessage}`, 'fail')
```

File: plugin-server/tests/worker/vm/upgrades/historical-export/export-historical-events-v2.test.ts [hunk 2]
```diff
@@ -583,6 +593,34 @@ describe('addHistoricalEventsExportCapabilityV2()', () => {
                 )
                 expect(await storage().get(EXPORT_PARAMETERS_KEY, null)).toEqual(null)
             })
+
+            it('stops export if abortMessage is set', async () => {
+                await storage().set(EXPORT_PARAMETERS_KEY, { ...params, abortMessage: 'test aborting' })
+
+                await coordinateHistoricalExport({
```
Comment (hazzadous) [inline plugin-server/tests/worker/vm/upgrades/historical-export/export-historical-events-v2.test.ts:600]: Can we call coordinateHistoricalExport again to verify that it doesn't try to stop the export again but rather just hits the shortcircuit for when EXPORT_PARAMETERS_KEY is null?

File: plugin-server/tests/worker/vm/upgrades/historical-export/export-historical-events-v2.test.ts [hunk 3]
```diff
@@ -370,6 +371,15 @@ describe('addHistoricalEventsExportCapabilityV2()', () => {
             expect(hub.db.queuePluginLogEntry).not.toHaveBeenCalled()
         })
 
+        it('stops export if abortMessage is set', async () => {
+            await storage().set(EXPORT_PARAMETERS_KEY, { ...exportParams, abortMessage: 'test aborting' })
+
+            await exportHistoricalEvents(defaultPayload)
+
+            expect(fetchEventsForInterval).not.toHaveBeenCalled()
+            expect(hub.db.queuePluginLogEntry).not.toHaveBeenCalled()
```
Comment (xvello) [inline plugin-server/tests/worker/vm/upgrades/historical-export/export-historical-events-v2.test.ts:380]: Should be called with a "Stopped due to aborted export" log

File: plugin-server/src/worker/vm/upgrades/historical-export/export-historical-events-v2.ts (hunk 1 above)
Comment (tiina303) [reply plugin-server/src/worker/vm/upgrades/historical-export/export-historical-events-v2.ts:258]: we might add a UI button in the future - this should be inside abortMessage

File: plugin-server/tests/worker/vm/upgrades/historical-export/export-historical-events-v2.test.ts (hunk 2 above)
Comment (tiina303) [reply plugin-server/tests/worker/vm/upgrades/historical-export/export-historical-events-v2.test.ts:600]: we have a test already for this case `does nothing if export isnt running / is done'`

**Review by xvello — APPROVED** [review]

**Review by tomasfarias — APPROVED** [review]
:shipit:

**Pull request merged**